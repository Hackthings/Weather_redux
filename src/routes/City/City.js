import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import SvgIcon from '../../components/SvgIcon';
import backArrowIcon from '../../resources/left-arrow.svg';
import Loader from '../../components/Loader/Loader';


const City = ({
    isRequestingInitialData,
    city,
    country,
    stateName,
    temperature,
    icon,
    iconUrl,
    localTime
}) => {
    return (
        <div>
            {isRequestingInitialData && (<Loader />)}
            {!isRequestingInitialData && (
                <div className="City">
                    <h1>{city} {country} {stateName}</h1>
                    <table className="description">
                        <tbody>
                            <tr>
                                <td>temperature: </td>
                                <td>{temperature}&deg; C</td>
                            </tr>
                            <tr>
                                <td>{icon}</td>
                                <td><img src={iconUrl} width="100"></img></td>
                            </tr>
                            <tr>
                                <td>local time:</td>
                                <td> {localTime}</td>
                            </tr>
                        </tbody>
                    </table>
                    <div className="City__iconWrapper">
                        <Link to='/'>
                            <SvgIcon src={backArrowIcon} className="City__iconWrapper__icon" />
                        </Link>
                    </div>
                </div>

            )}

        </div>
    )
}

export default City;