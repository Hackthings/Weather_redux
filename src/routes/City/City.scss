.City {
    margin: 0 auto;
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    h1 {
        font-size: 40px;
        padding: 30px 0;
        letter-spacing: 2px;
    }
    .description {
        border: 1px solid #ffffff;
        margin: 20px;
        border-collapse: collapse;
        font-family: 'Lato', sans-serif;
        tr {
            padding: 20px 0;
            td {
                font-size: 20px;
                padding: 20px;
                border: none;
                color: #12647f;
                text-align: left;
                background-color: #ffffff;
                &:nth-child(2) {
                    font-size: 24px;
                    font-weight: 700;
                }
            }
        }
    }
    &__iconWrapper {
        &__icon {
            svg {
                align-self: flex-start;
                width: 60px;
                path,
                polygon {
                    fill: #ffffff !important;
                }
            }
        }
    }
}