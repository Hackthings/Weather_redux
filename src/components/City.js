import React, { Component } from 'react';
import CityContainer from '../containers/CityContainer';


const City = (props) => {
    return (
        <CityContainer city={props.match.params.name}/>
    )
}

export default City;