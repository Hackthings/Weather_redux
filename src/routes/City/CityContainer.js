import React, { Component } from 'react';
import City from './City';

const url = "http://api.wunderground.com/api/4af47e77859b009e/conditions/q/";
const stateAbbreviation = 'CA';


class CityContainer extends Component {

    constructor(props) {
        super(props);

        this.state = {
            data: null,
            isRequestingInitialData: true
        }
    }

    componentDidMount() {
        const cityName = this.props.match.params.name;
        this.requestInitialData(cityName);
    }

    async requestInitialData(city) {
        const requestUrl = url + stateAbbreviation + '/' + city + '.json';
        this.setState({
            isRequestingInitialData: true
        });
        try {
            const response = await fetch(requestUrl);
            const parseResponse = await response.json();
            const parsed = this.parseResponse(parseResponse)

            console.log(parseResponse.current_observation)
            this.setState({
                ...parsed,
                isRequestingInitialData: false
            })
        }
        catch (error) {
            console.log('there was an error')
        }
    }

    parseResponse(response) {
        const {
            current_observation: {
                display_location: {
                    city,
                    country,
                    state_name
                },
                feelslike_c,
                icon,
                icon_url,
                local_time_rfc822
            }
        } = response;
        return {
            city,
            country,
            stateName: state_name,
            temperature: feelslike_c,
            icon,
            iconUrl: icon_url,
            localTime: local_time_rfc822
        }

    }

    render() {
        const {
            isRequestingInitialData,
            city,
            country,
            stateName,
            temperature,
            icon,
            iconUrl,
            localTime
        } = this.state;

        return (
            <City
                isRequestingInitialData={isRequestingInitialData}
                city={city}
                country={country}
                stateName={stateName}
                temperature={temperature}
                icon={icon}
                iconUrl={iconUrl}
                localTime={localTime}
            />
        )
    }


}

export default CityContainer;