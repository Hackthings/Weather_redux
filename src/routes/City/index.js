import './City.scss';

export {default} from './CityContainer'