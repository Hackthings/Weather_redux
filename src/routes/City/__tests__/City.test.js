import React from 'react';
import City from '../City';
import { shallow, mount } from 'enzyme';

describe('<City/>', () => {
    it('should render Loader', () => {
        const props = {
            isRequestingInitialData: true,
            city: 'test1',
            country: 'test2',
            stateName: 'test3',
            temperature: 'test4',
            icon: 'test5',
            iconUrl: 'test6',
            localTime: 'test7'
        }
        const wrapper = shallow(<City {...props} />);

        expect(wrapper).toMatchSnapshot('isRequestingInitialData=false');
        wrapper.setProps({isRequestingInitialData: false});
        expect(wrapper).toMatchSnapshot('isRequestingInitialData=true');

        
    })
})