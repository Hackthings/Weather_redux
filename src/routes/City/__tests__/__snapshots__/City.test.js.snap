// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<City/> should render Loader: isRequestingInitialData=false 1`] = `
<div>
  <Loader />
</div>
`;

exports[`<City/> should render Loader: isRequestingInitialData=true 1`] = `
<div>
  <div
    className="City"
  >
    <h1>
      test1
       
      test2
       
      test3
    </h1>
    <table
      className="description"
    >
      <tbody>
        <tr>
          <td>
            temperature: 
          </td>
          <td>
            test4
            Â° C
          </td>
        </tr>
        <tr>
          <td>
            test5
          </td>
          <td>
            <img
              src="test6"
              width="100"
            />
          </td>
        </tr>
        <tr>
          <td>
            local time:
          </td>
          <td>
             
            test7
          </td>
        </tr>
      </tbody>
    </table>
    <div
      className="City__iconWrapper"
    >
      <Link
        replace={false}
        to="/"
      >
        <Icon
          className="City__iconWrapper__icon"
          src="test-mock"
        />
      </Link>
    </div>
  </div>
</div>
`;
